### PR TITLE
user invites: Tell site admins when emails are not configured

### LINKED ISSUE #32308: user invites: Hint to site admins on how to set config mailing
Part of #31214

In #32270 we changed the null copy when emails aren't enabled. For server admins, we can use this real estate to show a helpful link to a documentation site on how to configure emails in their instance.

### PR DESCRIPTION
**Pull request by philipp-spiess**
Closes #32308
Depends on #32425

@rrhyne @slimsag Opening this PR so we can have a quick conversation on how to let admins know that emails aren't configured. Here's my first attempt. The link goes to: https://docs.sourcegraph.com/admin/config/site_config#email-smtp (which isn't super helpful at the moment either).

I also found https://github.com/sourcegraph/sourcegraph/issues/9017 so maybe we can make improvements to the docs as well?

## Test plan

Screenshot 2022-03-09 at 13 02 20

### CODE REVIEW

File: client/web/src/search/panels/CollaboratorsPanel.tsx [hunk 1]
```diff
@@ -174,7 +179,14 @@ const CollaboratorsPanelNullState: React.FunctionComponent<{ username: string }>
                 'h-100'
             )}
         >
-            {emailEnabled ? <div className="text-center">No collaborators found in sampled repositories.</div> : null}
+            {emailEnabled ? (
+                <div className="text-center">No collaborators found in sampled repositories.</div>
+            ) : isSiteAdmin ? (
+                <div className="text-center">
```
Comment (limitedmage) [inline client/web/src/search/panels/CollaboratorsPanel.tsx:185]: Should this text be muted to match the one below?

File: client/web/src/search/panels/CollaboratorsPanel.tsx [hunk 2]
```diff
@@ -174,7 +179,14 @@ const CollaboratorsPanelNullState: React.FunctionComponent<{ username: string }>
...
+                    This server is not configured to send emails.{' '}
+                    <Link to="https://docs.sourcegraph.com/admin/config/site_config#email-smtp">Learn more</Link>
```
Comment (limitedmage) [inline client/web/src/search/panels/CollaboratorsPanel.tsx:187]: This should use the `/help` redirect (see #32205)

**Review by limitedmage — APPROVED** [review]

File: client/web/src/search/panels/CollaboratorsPanel.tsx (hunk 1 above)
Comment (philipp-spiess) [reply client/web/src/search/panels/CollaboratorsPanel.tsx:185]: cc @rrhyne this is a question for you! I oriented myself on the Null screen for collaborators in this Figma: https://www.figma.com/file/Og1zVk7BbZ7SWTXM5WsWA5/Account-Setups-OKR-explorations?node-id=78%3A11652

**Comment by slimsag** [comment]
(merged `main` so as to add the doc page that was failing the CI check for broken links)

File: client/web/src/search/panels/CollaboratorsPanel.tsx (hunk 1 above)
Comment (rrhyne) [reply client/web/src/search/panels/CollaboratorsPanel.tsx:185]: It should match the others, my bad!

Comment (slimsag) [reply client/web/src/search/panels/CollaboratorsPanel.tsx]: Added text-muted per Rob's comment.

File: client/web/src/search/panels/CollaboratorsPanel.tsx [hunk 3]
```diff
@@ -174,7 +179,13 @@ const CollaboratorsPanelNullState: React.FunctionComponent<{ username: string }>
                 'h-100'
             )}
         >
-            {emailEnabled ? <div className="text-center">No collaborators found in sampled repositories.</div> : null}
+            {emailEnabled ? (
+                <div className="text-muted text-center">No collaborators found in sampled repositories.</div>
+            ) : isSiteAdmin ? (
+                <div className="text-muted text-center">
+                    This server is not configured to send emails. <Link to="/help/admin/config/email">Learn more</Link>
```
Comment (limitedmage) [inline client/web/src/search/panels/CollaboratorsPanel.tsx:186]: Minor accessibility issue: can we make this link more descriptive? For example: `See instruction on how to configure emails`. This helps users who navigate with a screen reader by tabbing through all the links.

**Pull request merged**